python_modules/dagster/dagster/_config/pythonic_config/__init__.py

```diff
@@ -891,17 +873,26 @@ def _with_updated_values(
         )
         return out
 
-    def _resolve_and_update_env_vars(self) -> "ConfigurableResourceFactory[TResValue]":
-        """Processes the config dictionary to resolve any EnvVar values. This is called at runtime
-        when the resource is initialized, so the user is only shown the error if they attempt to
-        kick off a run relying on this resource.
-
+    def _translate_processed_run_config_to_pydantic(
```

Comment:
s don't think this method name is accurate. We're creating a new copy of the pythonic resource sourced from the incoming run config.

`def _with_resource_config(self, resource_config)`